### PR TITLE
Backport Bitcoin 0.16 PRs which were backported into 0.15 - part1

### PR DESCRIPTION
**Pull request by codablock**
This is the first set of PRs which were backported into the Bitcoin 0.15 branch and then later released as minor 0.15.x releases. Instead of backporting the commits found in the 0.15 branch, I'm backporting the original PRs which were used to do the backports.

This PR currently builds on top of #3101. I'll rebase this PR later when this is merged.

### CODE REVIEW

File: test/functional/rawtransactions.py [hunk 1]
```diff
@@ -124,6 +124,55 @@ def run_test(self):
...
+        self.log.info(rawTxPartialSigned2)
+        assert_equal(rawTxPartialSigned2['complete'], False) #node2 only has one key, can't comp. sign the tx
+        rawTxComb = self.nodes[2].combinerawtransaction([rawTxPartialSigned1['hex'], rawTxPartialSigned2['hex']])
+        self.log.info(rawTxComb)
+        self.nodes[2].sendrawtransaction(rawTxComb)
+        rawTx2 = self.nodes[0].decoderawtransaction(rawTxComb)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), bal+Decimal('50.00000000')+Decimal('2.19000000')) #block reward + tx
```
Comment (UdjinM6) [inline test/functional/rawtransactions.py]:
```suggestion
        assert_equal(self.nodes[0].getbalance(), bal+Decimal('500.00000000')+Decimal('2.19000000')) #block reward + tx
```

**Comment by codablock** [comment]
Rebased on develop to make sure this PR still works with the bunch of changes which happened on develop

**Comment by UdjinM6** [comment]
`llmq-is-cl-conflicts.py` failed, should be fixed by #3094

**Comment by codablock** [comment]
Rebased on develop to make Travis happy

**Pull request merged**